yarn-project/noir-compiler/src/cli/contract.ts

```diff
@@ -30,20 +36,22 @@ export function compileContract(program: Command, name = 'contract', log: LogFn
           outdir: string;
           typescript: string | undefined;
           interface: string | undefined;
+          compiler: string | undefined;
         },
         /* eslint-enable jsdoc/require-jsdoc */
       ) => {
-        const { outdir, typescript, interface: noirInterface } = options;
+        const { outdir, typescript, interface: noirInterface, compiler = 'nargo' } = options;
         if (typeof projectPath !== 'string') throw new Error(`Missing project path argument`);
         const currentDir = process.cwd();
 
-        const compile = compileUsingNargo;
+        const compile = compiler === 'wasm' ? compileUsingNoirWasm : compileUsingNargo;
```

Comment:
Let's validate that `compiler` is one of `nargo` or `wasm` rather than silently going with nargo if the user makes a typo.